lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -568,6 +568,11 @@ class TTIRLayoutDPSOperandsRewriter
     for (auto &operand : op->getOpOperands()) {
       bool isResult = op.isDpsInit(&operand);
 
+      // @lpanos : TTNN Conv2d moves input, weight, and bias from host to device
```

Comment:
Can you remove your username?

I think it's also only true for weight and bias.

Reply:
So in `third_party/tt-mlir/third_party/tt-metal/src/tt-metal/ttnn/cpp/ttnn/operations/conv2d/conv2d.cpp` line 417 you'll see an if-else block that handles the input tensor differently whether or not it is on device. I've tried enabling the movement of the input tensor to device pre-conv2d, and I am seeing data mismatches on the output. I'm unsure why exactly this is. The TTNN test in `third_party/tt-mlir/third_party/tt-metal/src/tt-metal/tests/ttnn/unit_tests/operations/test_new_conv2d.py` looks like it does not move the tensors to device before executing `conv2d` either...

Reply:
The memory config op placed on the output of the first conv moves it to host I believe.